hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDDV2.scala

```diff
@@ -142,6 +142,9 @@ class HoodieMergeOnReadRDDV2(@transient sc: SparkContext,
         val properties = TypedProperties.fromMap(options.asJava)
         properties.setProperty(MAX_MEMORY_FOR_MERGE.key(), String.valueOf(maxCompactionMemoryInBytes))
         properties.setProperty(HoodieReaderConfig.MERGE_TYPE.key(), mergeType)
+        if (!properties.containsKey(HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.key())) {
+          properties.put(HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.key(), HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.defaultValue())
```

Comment:
we can get rid of this if we have the props fallback fetch.

Reply:
will remove this line.

Reply:
I think we might not be able to do this props fallback fetch in builder https://github.com/apache/hudi/pull/13591#discussion_r2226630154

Reply:
will fix based on our discussion.